src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/ReflectionInvokeMapNode.cs

```diff
@@ -111,6 +91,26 @@ static void AddSignatureDependency(ref DependencyList dependencies, NodeFactory
             ReflectionVirtualInvokeMapNode.GetVirtualInvokeMapDependencies(ref dependencies, factory, method);
         }
 
+        internal static void AddSignatureDependency(ref DependencyList dependencies, NodeFactory factory, TypeDesc type, string reason)
+        {
+            if (type.IsByRef)
+                type = ((ParameterizedType)type).ParameterType;
+
+            // Pointer runtime type handles can be created at runtime if necessary
+            while (type.IsPointer)
+                type = ((ParameterizedType)type).ParameterType;
+
+            // Skip tracking dependencies for primitive types. Assume that they are always present.
+            if (type.IsPrimitive || type.IsVoid)
+                return;
+
+            TypeDesc canonType = type.ConvertToCanonForm(CanonicalFormKind.Specific);
+            if (canonType.IsCanonicalSubtype(CanonicalFormKind.Any))
+                GenericTypesTemplateMap.GetTemplateTypeDependencies(ref dependencies, factory, type.ConvertToCanonForm(CanonicalFormKind.Specific));
```

Comment:
Can this be
```csharp
GenericTypesTemplateMap.GetTemplateTypeDependencies(ref dependencies, factory, canonType);
```

We [just computed the `canonType`](https://github.com/dotnet/runtime/pull/88083/files#diff-3dcbf6471d7684e53a20ea7c269edd4789596d7f89132db9a1086a552a3f63d0R107)

Reply:
Yes, it can, good catch.

Reply:
Would you be willing to submit a PR?

Reply:
For one line, sure... :)

Reply:
Done https://github.com/dotnet/runtime/pull/88171